src/neo/Oracle/OraclePolicyContract.cs

```diff
@@ -0,0 +1,158 @@
+using Neo.Cryptography.ECC;
+using Neo.IO;
+using Neo.Ledger;
+using Neo.Network.P2P.Payloads;
+using Neo.Persistence;
+using Neo.SmartContract;
+using Neo.SmartContract.Manifest;
+using Neo.SmartContract.Native;
+using Neo.VM;
+using Neo.VM.Types;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using Array = Neo.VM.Types.Array;
+
+namespace Neo.Oracle
+{
+    public sealed class OraclePolicyContract : NativeContract
+    {
+        public override string ServiceName => "Neo.Native.Oracle.Policy";
+
+        public override int Id => -4;
+
+        private const byte Prefix_Validator = 24;
+        private const byte Prefix_TimeOutMilliSeconds = 25;
+        private const byte Prefix_PerRequestFee = 26;
+
+        public OraclePolicyContract()
+        {
+            Manifest.Features = ContractFeatures.HasStorage;
+        }
+
+        internal override bool Initialize(ApplicationEngine engine)
+        {
+            if (!base.Initialize(engine)) return false;
+            engine.Snapshot.Storages.Add(CreateStorageKey(Prefix_TimeOutMilliSeconds), new StorageItem
+            {
+                Value = BitConverter.GetBytes(1000)
+            });
+            engine.Snapshot.Storages.Add(CreateStorageKey(Prefix_PerRequestFee), new StorageItem
+            {
+                Value = BitConverter.GetBytes(1000)
+            });
+            return true;
+        }
+
+        [ContractMethod(0_01000000, ContractParameterType.Boolean, ParameterTypes = new[] { ContractParameterType.Hash160, ContractParameterType.Array }, ParameterNames = new[] { "account", "pubkeys" })]
+        private StackItem DelegateOracleValidator(ApplicationEngine engine, Array args)
+        {
+            UInt160 account = new UInt160(args[0].GetSpan());
+            if (!InteropService.Runtime.CheckWitnessInternal(engine, account)) return false;
+            ECPoint[] pubkeys = ((Array)args[1]).Select(p => p.GetSpan().AsSerializable<ECPoint>()).ToArray();
+            if (pubkeys.Length != 2) return false;
+            StoreView snapshot = engine.Snapshot;
+            StorageKey key = CreateStorageKey(Prefix_Validator, pubkeys[0]);
+            if (snapshot.Storages.TryGet(key) != null)
```

Comment:
Should check `pubkeys[0]` wether is one of the CNs?

Reply:
As we discussed, it is not necessary.

Reply:
Sorry @doubiliu can you link the discussion, I also think we are missing
```csharp
UInt160 account = GetOracleMultiSigAddress(snapshot);
if (!InteropService.Runtime.CheckWitnessInternal(engine, account)) return false;
```

Reply:
Also an array with only one entry it's easier extend in the future.

Reply:
It's just a temporary conversation.Because the CN node changes dynamically, once the CN node changes, the original check will be meaningless.We also have to do special logic for CN changes, which brings a very invasive nature to OraclePolicy. Therefore, I designed it as a public method, and any node can perform delegate operations, just filtering when taking OracleValidator

仅仅是一个临时对话。因为CN节点是动态变化的，一旦CN节点变更，原先的检查就会失去意义。我们还得针对CN变更做专门的逻辑，这给OraclePolicy带来了很强的侵入性。因此，我设计成了公开的方法，任何节点都可以进行委托操作，只是取OracleValidator时做过滤